<launch>

  <!-- Args corresponding to environment variables -->
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <!-- Set to false if the nodelet should run under an external nodelet manager -->
  <arg name="standalone" default="true" />

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value="" />
  <arg if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch" />

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="manager" default="mask_picker_manager" />
  <arg if="$(arg standalone)" name="nodelet" value="standalone" />
  <arg unless="$(arg standalone)" name="nodelet" value="load" />
  <arg if="$(arg standalone)" name="nodelet_manager" value="" />
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)" />



<!-- Mask Picker nodelet -->
<node pkg="nodelet" type="nodelet" name="mask_picker" args="$(arg nodelet) cable_analysis/MaskPicker $(arg nodelet_manager)"
    launch-prefix="$(arg launch_prefix_debug)" output="screen">


    <!-- ROS parameters config files -->
    <rosparam file="$(find cable_analysis)/config/default.yaml" />

    <!-- Subscribers -->
    <remap from="~camera_color_in" to="/camera/color/image_raw" />

</node>

</launch>